src/features/code_actions.zig

```diff
@@ -166,64 +164,67 @@ fn handleUnusedVariableOrConstant(builder: *Builder, actions: *std.ArrayListUnma
     });
 }
 
-fn handleUnusedIndexCapture(builder: *Builder, actions: *std.ArrayListUnmanaged(types.CodeAction), loc: offsets.Loc) !void {
+fn handleUnusedCapture(builder: *Builder, actions: *std.ArrayListUnmanaged(types.CodeAction), loc: offsets.Loc) !void {
     const tracy_zone = tracy.trace(@src());
     defer tracy_zone.end();
 
-    const capture_locs = getCaptureLoc(builder.handle.text, loc, true) orelse return;
+    const capture_loc = getCaptureLoc(builder.handle.text, loc) orelse return;
 
-    // TODO support discarding without modifying the capture
-    // by adding a discard in the block scope
-    const is_value_discarded = std.mem.eql(u8, offsets.locToSlice(builder.handle.text, capture_locs.value), "_");
-    if (is_value_discarded) {
-        // |_, i| ->
-        // TODO fix formatting
-        try actions.append(builder.arena, .{
-            .title = "remove capture",
-            .kind = .quickfix,
-            .isPreferred = true,
-            .edit = try builder.createWorkspaceEdit(&.{builder.createTextEditLoc(capture_locs.loc, "")}),
-        });
-    } else {
-        // |v, i| -> |v|
-        // |v, _| -> |v|
-        try actions.append(builder.arena, .{
-            .title = "remove index capture",
-            .kind = .quickfix,
-            .isPreferred = true,
-            .edit = try builder.createWorkspaceEdit(&.{builder.createTextEditLoc(
-                .{ .start = capture_locs.value.end, .end = capture_locs.loc.end - 1 },
-                "",
-            )}),
-        });
+    // look for next non-whitespace after last '|'. if its a '{' we can insert discards.
+    // this means bare loop/switch captures (w/out curlies) aren't supported.
+    var block_start = capture_loc.end + 1;
+    while (block_start < builder.handle.text.len and
+        std.ascii.isWhitespace(builder.handle.text[block_start])) : (block_start += 1)
+    {}
+    if (builder.handle.text[block_start] != '{') {
```

Comment:
This breaks for a case like
```
    for (0..10, 0..10, 0..10) |i, j, k| i = 1;
```
but so does every other capture, definitely doesn't have to be fixed in this PR; perhaps we could open an issue for this :)

Reply:
created https://github.com/zigtools/zls/issues/1203